### ISSUE TITLE
Pure AES implementation

### ISSUE THREAD

**Issue by notgne2** [Ragnaroek/mithril#16]
Is there going to be a pure/Rust-only (no embedded assembly) option for AES?

**Comment by Ragnaroek** [comment]
I think not in the near future, because performance would be _very_ slow with a pure software AES implementation.

The nice thing would be that the miner works on all platform Rust can be compiled to, but this would still be slow.

Next on the roadmap is proabably ARM support. Are you asking with that architecture in mind?

**Comment by DuckerMan** [comment]
@Ragnaroek why mithril can't use [AES](https://docs.rs/aes/0.5.0/aes/) crate? It supports [Hardware AES](https://docs.rs/aesni/0.9.0/aesni/).

**Comment by Ragnaroek** [comment]
@DuckerMan Yes, that would be an option. Maybe this crate didn't exist when mithril was created? But I will first implement RandomX before considering this.
